collections/concurrent_map.go

```diff
@@ -0,0 +1,69 @@
+package collections
+
+import "sync"
+
+type ConcurrentMap[Key comparable, Value any] struct {
+	mtx sync.RWMutex
+	m   map[Key]Value
+}
+
+func NewConcurrentMap[Key comparable, Value any]() *ConcurrentMap[Key, Value] {
+	return &ConcurrentMap[Key, Value]{
+		m: make(map[Key]Value),
+	}
+}
+
+func (cm *ConcurrentMap[Key, Value]) Add(key Key, val Value) {
```

Comment:
nit: the standard naming for this is usually `Set`
```suggestion
func (cm *ConcurrentMap[Key, Value]) Set(key Key, val Value) {
```